### PR TITLE
Create matcher

### PR DESCRIPTION
**Pull request by pmarsh-scottlogic**
created and tested the matcher

### CODE REVIEW

File: src/main/java/com/example/demo/matcher/Matcher.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Matcher {
+    private final OrderService orderService;
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/Matcher.java:18]: I might be wrong but could these be `@Autowired`? I would have a read about this here: https://www.baeldung.com/inversion-control-and-dependency-injection-in-spring . I think the best practice is constructor injection but I might be wrong. Let me know what you think about this :)

File: src/main/java/com/example/demo/matcher/models/OrderbookItem.java [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 import lombok.AllArgsConstructor;
 import lombok.Getter;
-import lombok.Setter;
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/models/OrderbookItem.java:5]: Good job getting rid of it here 👍

File: src/test/java/com/example/demo/MatcherTests.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+@SpringBootTest
+public class MatcherTests {
+    Matcher matcher;
+    OrderService orderService;
+    TradeService tradeService;
+
+    @BeforeEach
+    void initialise() {
+        orderService = Mockito.mock(OrderService.class);
```
Comment (nataliadombrowska) [inline src/test/java/com/example/demo/MatcherTests.java:30]: Nice use of mock here 👍

File: src/test/java/com/example/demo/TestUtils.java [hunk 4]
```diff
@@ -37,4 +38,15 @@ public static UUID uuidFromString(String s) {
     public static OrderbookItem makeOrderbookItem(float price, float quantity) {
         return new OrderbookItem(new BigDecimal(price), new BigDecimal(quantity));
     }
+
+    public static Trade randomTrade() {
```
Comment (nataliadombrowska) [inline src/test/java/com/example/demo/TestUtils.java]: Good job using a random trades for testing!

**Review by nataliadombrowska — CHANGES_REQUESTED** [review]
Great job! Just one little thing to consider but apart from that I'm happy with the code 👍

File: src/main/java/com/example/demo/matcher/models/OrderbookItem.java (hunk 2 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/matcher/models/OrderbookItem.java:5]: IntelliJ has great tools for tidying code!

File: build.gradle [hunk 5]
```diff
@@ -25,3 +26,14 @@ dependencies {
 tasks.named('test') {
 	useJUnitPlatform()
 }
+
+jacoco {
```
Comment (nataliadombrowska) [inline build.gradle:30]: Nice! I've never heard of this library but its definitely good to see code coverage 👍

File: src/main/java/com/example/demo/matcher/Matcher.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    private final TradeService tradeService;
+
+    @Autowired
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/Matcher.java:21]: Good stuff :) Thanks for adding that in!

**Review by nataliadombrowska — APPROVED** [review]
Good job!

**Pull request merged**